clar2wasm/src/tools.rs

```diff
@@ -73,9 +73,65 @@ impl TestEnvironment {
     }
 
     pub fn new(epoch: StacksEpochId, version: ClarityVersion) -> Self {
+        let (epoch, version) = if !Self::epoch_and_clarity_match(epoch, version) {
+            let valid_version = ClarityVersion::default_for_epoch(epoch);
+            println!(
+                "[WARN] Provided epoch ({epoch}) and Clarity version ({version}) do not match. \
+                 Defaulting to epoch ({epoch}) and Clarity version ({valid_version})."
+            );
+            (epoch, valid_version)
+        } else {
+            (epoch, version)
+        };
+
         Self::new_with_amount(1_000_000_000, epoch, version)
     }
 
+    /// Checks whether the given epoch and Clarity version are compatible.
+    ///
+    /// # Parameters
+    ///
+    /// - `epoch`: The `StacksEpochId` representing the current epoch.
+    /// - `version`: The `ClarityVersion` representing the Clarity version to check.
+    ///
+    /// # Returns
+    ///
+    /// Returns `true` if the specified `epoch` supports the given `ClarityVersion`,
+    /// and `false` otherwise.
+    ///
+    pub fn epoch_and_clarity_match(epoch: StacksEpochId, version: ClarityVersion) -> bool {
+        match (epoch, version) {
+            // For Epoch10, no clarity version is supported.
+            (StacksEpochId::Epoch10, _) => false,
+
+            // For epochs 20, 2_05, 21, 22, 23, 24, and 25,
+            // Clarity1 and Clarity2 are supported but Clarity3 is not.
+            (
+                StacksEpochId::Epoch20
+                | StacksEpochId::Epoch2_05
+                | StacksEpochId::Epoch21
+                | StacksEpochId::Epoch22
+                | StacksEpochId::Epoch23
+                | StacksEpochId::Epoch24
+                | StacksEpochId::Epoch25,
+                ClarityVersion::Clarity3,
+            ) => false,
+            (
+                StacksEpochId::Epoch20
+                | StacksEpochId::Epoch2_05
+                | StacksEpochId::Epoch21
+                | StacksEpochId::Epoch22
+                | StacksEpochId::Epoch23
+                | StacksEpochId::Epoch24
+                | StacksEpochId::Epoch25,
+                _,
+            ) => true,
```

Comment:
It would be closer to the comment to have something like
```suggestion
            // For epochs 20, 2_05, 21, 22, 23, 24, and 25,
            // Clarity1 and Clarity2 are supported but Clarity3 is not.
            (
                StacksEpochId::Epoch20
                | StacksEpochId::Epoch2_05
                | StacksEpochId::Epoch21
                | StacksEpochId::Epoch22
                | StacksEpochId::Epoch23
                | StacksEpochId::Epoch24
                | StacksEpochId::Epoch25,
                version,
            ) => version != ClarityVersion::Clarity3,
```
(and if `ClarityVersion` implements `Ord`, using `version <= ClarityVersion::Clarity2` would be even better).

Reply:
Thanks for the suggestion! 👍 

I haven't commited your code to use the Ordering comparison.